parboiled-core/src/test/scala/org/parboiled2/TestParserSpec.scala

```diff
@@ -16,34 +16,43 @@
 
 package org.parboiled2
 
-import org.specs2.specification.Scope
-import org.specs2.specification.dsl.NoReferenceDsl
-import org.specs2.mutable.Specification
-import org.specs2.control.NoNumberOfTimes
 import org.parboiled2.support.Unpack
 import shapeless._
+import utest._
 
-abstract class TestParserSpec extends Specification with NoReferenceDsl with NoNumberOfTimes {
+abstract class TestParserSpec extends TestSuite {
   type TestParser0 = TestParser[HNil, Unit]
   type TestParser1[T] = TestParser[T :: HNil, T]
   type TestParserN[L <: HList] = TestParser[L, L]
 
-  // work-around for https://github.com/etorreborre/specs2/issues/514
-  override def mutableLinkFragment(alias: String): mutableLinkFragment = ???
-  override def mutableSeeFragment(alias: String): mutableSeeFragment = ???
-
-  abstract class TestParser[L <: HList, Out](implicit unpack: Unpack.Aux[L, Out]) extends Parser with Scope {
+  abstract class TestParser[L <: HList, Out](implicit unpack: Unpack.Aux[L, Out]) extends Parser {
     var input: ParserInput = _
     def errorFormatter: ErrorFormatter = new ErrorFormatter(showTraces = true)
 
     def targetRule: RuleN[L]
 
-    def beMatched = beTrue ^^ (parse(_: String).isRight)
-    def beMatchedWith(r: Out) = parse(_: String) === Right(r)
-    def beMismatched = beTrue ^^ (parse(_: String).isLeft)
-    def beMismatchedWithError(pe: ParseError) = parse(_: String).left.toOption.get === pe
-    def beMismatchedWithErrorMsg(msg: String) =
-      parse(_: String).left.toOption.map(formatError(_, errorFormatter)).get === msg.stripMargin
+
+    sealed trait MustAssert{
```

Comment:
Nicely done, this small mini-DSL to maintain the nice readability of the parser tests!